openhands/llm/llm.py

```diff
@@ -122,27 +122,6 @@ def __init__(
             drop_params=self.config.drop_params,
         )
 
-        if self.vision_is_active():
-            logger.debug('LLM: model has vision enabled')
-        if self.is_caching_prompt_active():
-            logger.debug('LLM: caching prompt enabled')
-        if self.is_function_calling_active():
-            logger.debug('LLM: model supports function calling')
-
-        self._completion = partial(
```

Comment:
This assignment is not duplicate, I don't think? If you feel it is, could you point out where is the other?

The logging lines are duplicate, and it's nice to clean them out.

Reply:
https://github.com/All-Hands-AI/OpenHands/blob/main/openhands/llm/llm.py#L111
https://github.com/All-Hands-AI/OpenHands/blob/main/openhands/llm/llm.py#L132
this two lines do the same assignment, I dont know why

Reply:
Oh, you're right, this is so weird. I have a guess... the init model info call is necessary for the variables assigned in the partial, some wires must have got crossed because of that.

There's no need for an issue, sorry.

Reply:
Could you please just check this, the order, that we need initialize model info at the right time?

Reply:
> What do you think about this hypothesis - could you please check the order, e.g. that we need initialize model info at the right time?

I checked the init phase in self.init_model_info(), didn't see any related variables that would be used in the partial invocation. BTW, I don't think partial would change any variable which is outside of partial code scope.
We can ask @rbren for confirmation, he submitted this commit.